Remove document.implementation.createCSSStyleSheet()

karlcow opened the pull request:
#### 7a15673a3829331aedb8978ca73400e528f09b39
<pre>
Remove document.implementation.createCSSStyleSheet()
<a href="https://bugs.webkit.org/show_bug.cgi?id=285157">https://bugs.webkit.org/show_bug.cgi?id=285157</a>
<a href="https://rdar.apple.com/142045904">rdar://142045904</a>

The code has never been implemented in Firefox
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=63850">https://bugzilla.mozilla.org/show_bug.cgi?id=63850</a>

This was part of DOM Level 2
<a href="https://www.w3.org/TR/2000/REC-DOM-Level-2-Style-20001113/css.html#CSS-DOMImplementationCSS">https://www.w3.org/TR/2000/REC-DOM-Level-2-Style-20001113/css.html#CSS-DOMImplementationCSS</a>

Safari, Firefox and Chrome returns undefined
document.implementation.createCSSStyleSheet

* LayoutTests/fast/css/DOMImplementation.createCSSStyleSheet-obsolete-expected.txt: Removed.
* LayoutTests/fast/css/DOMImplementation.createCSSStyleSheet-obsolete.html: Removed.
* LayoutTests/platform/mac-site-isolation/TestExpectations:
* LayoutTests/webgl/2.0.0/resources/webgl_test_files/deqp/temp_externs/w3c_css.js:
(DOMImplementationCSS.prototype.createCSSStyleSheet): Deleted.
* Source/WebCore/dom/DOMImplementation.cpp:
(WebCore::DOMImplementation::createCSSStyleSheet): Deleted.
* Source/WebCore/dom/DOMImplementation.h:
* Source/WebInspectorUI/UserInterface/Models/NativeFunctionParameters.js:
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMDOMImplementation.cpp:
(webkit_dom_dom_implementation_create_css_style_sheet): Deleted.
* Source/WebKitLegacy/mac/DOM/DOMImplementation.h:
* Source/WebKitLegacy/mac/DOM/DOMImplementation.mm:
(-[DOMImplementation createCSSStyleSheet:media:]): Deleted.
(-[DOMImplementation createCSSStyleSheet::]): Deleted.
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7a15673a3829331aedb8978ca73400e528f09b39

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/82415 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/2079 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/36505 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/87547 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/33476 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/84520 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/2150 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/9966 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/64237 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/21989 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/85484 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/1527 "Found 1 new test failure: fast/forms/ios/file-upload-panel.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/74988 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/44514 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/1427 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/29173 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/32517 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/72718 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/29807 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/88903 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/9721 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/6983 "Found 1 new test failure: http/wpt/mediarecorder/record-96KHz-sources.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/72636 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/9947 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/70802 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/71853 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/15994 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/15000 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/1094 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/9674 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/15195 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/9548 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/13014 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/11318 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov reviewed (changes requested):
We cannot modify released benchmarks.

Cf. https://bugs.webkit.org/show_bug.cgi?id=171825

karlcow committed:
Remove document.implementation.createCSSStyleSheet()
https://bugs.webkit.org/show_bug.cgi?id=285157
rdar://142045904

The code has never been implemented in Firefox
https://bugzilla.mozilla.org/show_bug.cgi?id=63850

This was part of DOM Level 2
https://www.w3.org/TR/2000/REC-DOM-Level-2-Style-20001113/css.html#CSS-DOMImplementationCSS

Safari, Firefox and Chrome returns undefined
document.implementation.createCSSStyleSheet

* LayoutTests/fast/css/DOMImplementation.createCSSStyleSheet-obsolete-expected.txt: Removed.
* LayoutTests/fast/css/DOMImplementation.createCSSStyleSheet-obsolete.html: Removed.
* LayoutTests/platform/mac-site-isolation/TestExpectations:
* LayoutTests/webgl/2.0.0/resources/webgl_test_files/deqp/temp_externs/w3c_css.js:
(DOMImplementationCSS.prototype.createCSSStyleSheet): Deleted.
* Source/WebCore/dom/DOMImplementation.cpp:
(WebCore::DOMImplementation::createCSSStyleSheet): Deleted.
* Source/WebCore/dom/DOMImplementation.h:
* Source/WebInspectorUI/UserInterface/Models/NativeFunctionParameters.js:
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMDOMImplementation.cpp:
(webkit_dom_dom_implementation_create_css_style_sheet): Deleted.
* Source/WebKitLegacy/mac/DOM/DOMImplementation.h:
* Source/WebKitLegacy/mac/DOM/DOMImplementation.mm:
(-[DOMImplementation createCSSStyleSheet:media:]): Deleted.
(-[DOMImplementation createCSSStyleSheet::]): Deleted.

annevk commented on Source/WebKitLegacy/mac/DOM/DOMImplementation.mm:
```diff
@@ -105,11 +99,6 @@ - (DOMDocument *)createDocument:(NSString *)namespaceURI :(NSString *)qualifiedN
     return [self createDocument:namespaceURI qualifiedName:qualifiedName doctype:doctype];
 }
 
-- (DOMCSSStyleSheet *)createCSSStyleSheet:(NSString *)title :(NSString *)media
-{
-    return [self createCSSStyleSheet:title media:media];
-}
```
Instead of removing all this, use "return nil" as the body instead.

annevk replied:
Maybe you can remove some of the includes.
